src/fw/drivers/sf32lb52/watchdog.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 Core Devices LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/watchdog.h"
+
+#include "util/bitset.h"
+#include "system/logging.h"
+#include <inttypes.h>
+#include "bf0_hal.h"
+
+
+static WDT_HandleTypeDef hwdt = {
+    .Instance = hwp_wdt1,
+};
+static WDT_HandleTypeDef hiwdt = {
+    .Instance = hwp_iwdt,
+};
+
+#define IWDT_RELOAD_DIFFTIME 5
+#define WDT_RELOADER_TIMEOUT 3
+#define WDT_REBOOT_TIMEOUT 50
```

Comment:
In practice, we normally put macro define before static local variable, please adjust the order.
And the reboot timeout is about 8 seconds.